File: flutter-im-ui/lib/src/components/flare_screen.dart
import 'package:flutter/material.dart';

import '../tokens/flare_tokens.dart';

/// Page surface for [FlareScreen] — mirrors the Vue contract.
enum FlareScreenSurface { canvas, surface, aurora }

/// FlareScreen — the base page scaffold every Flare business page builds on.
///
/// It owns the themed page surface (auto light/dark — the app's [ThemeMode]
/// drives it, so `ThemeMode.system` follows the OS), an optional header
/// (back / large title / actions), and a scrollable, safe-area body. Business
/// code writes `FlareScreen(title: '设置', onBack: pop, child: …)` and gets a
/// consistent, fully themeable page — no page-level colours are hard-coded.
class FlareScreen extends StatelessWidget {
  const FlareScreen({
    super.key,
    this.title,
    this.onBack,
    this.actions,
    this.surface = FlareScreenSurface.canvas,
    this.padded = false,
    this.scroll = true,
    this.footer,
    required this.child,
  });

  /// Large-title text. Omit for a headerless page.
  final String? title;

  /// When non-null, a leading back button is shown and calls this.
  final VoidCallback? onBack;

  /// Trailing header widgets (icons / buttons).
  final List<Widget>? actions;

  /// `canvas` grouped-list bg (default), `surface` flat panel, `aurora` violet wash.
  final FlareScreenSurface surface;

  /// Pad the body (16px).
  final bool padded;

  /// Scrollable body. Default true.
  final bool scroll;

  /// Optional pinned footer.
  final Widget? footer;

  final Widget child;

  @override
  Widget build(BuildContext context) {
    final colors = FlareColors.of(Theme.of(context).brightness);
    final base = surface == FlareScreenSurface.surface ? colors.bgPrimary : colors.bgSecondary;
    final hasHeader = title != null || onBack != null || (actions?.isNotEmpty ?? false);

    Widget body = padded
        ? Padding(padding: const EdgeInsets.all(FlareSizes.spacingLg), child: child)
        : child;
    if (scroll) body = SingleChildScrollView(child: body);

    return DecoratedBox(
      decoration: BoxDecoration(
        color: surface == FlareScreenSurface.aurora ? null : base,
        // Aurora — a soft violet light wash at the top of the canvas.
        gradient: surface == FlareScreenSurface.aurora
            ? LinearGradient(
                begin: Alignment.topCenter,
                end: Alignment.center,
                colors: [
                  Color.alphaBlend(colors.primary.withValues(alpha: 0.18), base),
                  base,
                ],
              )
            : null,
      ),
      child: SafeArea(
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.stretch,
          children: [
            if (hasHeader)
              Padding(
                padding: const EdgeInsets.fromLTRB(FlareSizes.spacingMd,
                    FlareSizes.spacingSm, FlareSizes.spacingMd, FlareSizes.spacingSm),
                child: Row(
                  children: [
                    if (onBack != null)
                      IconButton(
                        icon: Icon(Icons.arrow_back, color: colors.textPrimary),
                        onPressed: onBack,
                        padding: EdgeInsets.zero,
                        constraints: const BoxConstraints(minWidth: 40, minHeight: 40),
                      ),
                    if (title != null)
                      Expanded(
                        child: Text(title!,
                            maxLines: 1,
                            overflow: TextOverflow.ellipsis,
                            style: TextStyle(
                                color: colors.textPrimary, fontSize: 24, fontWeight: FontWeight.bold)),
                      )
                    else
                      const Spacer(),
                    ...?actions,
                  ],
                ),
              ),
            Expanded(child: body),
            if (footer != null) footer!,
          ],
        ),
      ),
    );
  }
}

File: android-im-ui/src/main/kotlin/com/flare/im/ui/FlareScreen.kt
package com.flare.im.ui

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.outlined.ArrowBack
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.lerp
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

enum class FlareScreenSurface { Canvas, Surface, Aurora }

/**
 * FlareScreen — the base page scaffold for Flare business pages.
 *
 * Themed page surface (auto light/dark via [flareColors], which reads
 * `isSystemInDarkTheme()` unless the app forces a theme), optional header
 * (back / large title / actions), and a scrollable body. Mirrors the Vue
 * contract — business code writes `FlareScreen(title = "设置", onBack = pop) { … }`
 * and gets a consistent, fully themeable page with no hard-coded page colours.
 */
@Composable
fun FlareScreen(
    title: String? = null,
    onBack: (() -> Unit)? = null,
    surface: FlareScreenSurface = FlareScreenSurface.Canvas,
    padded: Boolean = false,
    scroll: Boolean = true,
    actions: (@Composable () -> Unit)? = null,
    content: @Composable () -> Unit,
) {
    val colors = flareColors()
    val base = if (surface == FlareScreenSurface.Surface) colors.bgPrimary else colors.bgSecondary
    val hasHeader = title != null || onBack != null || actions != null

    val bg = if (surface == FlareScreenSurface.Aurora) {
        // Aurora — a soft violet light wash at the top of the canvas.
        Modifier.background(Brush.verticalGradient(listOf(lerp(colors.primary, base, 0.84f), base, base)))
    } else {
        Modifier.background(base)
    }

    Column(Modifier.fillMaxSize().then(bg)) {
        if (hasHeader) {
            Row(
                Modifier.fillMaxWidth().padding(horizontal = FlareSizes.spacingMd, vertical = FlareSizes.spacingSm),
                verticalAlignment = Alignment.CenterVertically,
            ) {
                if (onBack != null) {
                    IconButton(onClick = onBack, modifier = Modifier.size(40.dp)) {
                        Icon(Icons.AutoMirrored.Outlined.ArrowBack, flareStrings().back, tint = colors.textPrimary)
                    }
                }
                if (title != null) {
                    Text(
                        title, color = colors.textPrimary, fontSize = 24.sp, fontWeight = FontWeight.Bold,
                        maxLines = 1, overflow = TextOverflow.Ellipsis, modifier = Modifier.weight(1f),
                    )
                } else {
                    Spacer(Modifier.weight(1f))
                }
                actions?.invoke()
            }
        }
        Column(
            Modifier.fillMaxWidth().weight(1f)
                .then(if (scroll) Modifier.verticalScroll(rememberScrollState()) else Modifier)
                .then(if (padded) Modifier.padding(FlareSizes.spacingLg) else Modifier),
        ) {
            content()
        }
    }
}
